File: backend/packages/projects/src/routes/create.ts
import { BaseRoute } from '../utils/route'
import { BaseService } from '../utils/service'
import { makeResponse } from '../utils/response'

type Input = {
    connectionId: string
    name: string
    description: string
}

interface CreateRoute extends BaseRoute<Input, any> { }

type Self = {
    service: BaseService<Input, any>
}

export const makeCreateRoute = (service: BaseService<Input, any>): CreateRoute => {
    const self: Self = {
        service
    }

    return {
        call: call(self)
    }
}

const call = (self: Self): CreateRoute['call'] => async (request) => {
    const connectionId = request.data.connectionId
    const name = request.data.name
    const description = request.data.description

    const response = await self.service.call({ connectionId, name, description })

    return makeResponse(
        200,
        request.action,
        request.requestId,
        response
    )
}

File: backend/packages/users/src/routes/create.py
from ..utils.route import BaseRoute
from ..utils.service import BaseService
from ..utils.request import Request
from ..utils.response import Response


class CreateRoute(BaseRoute):
    _service: BaseService

    def __init__(self, service: BaseService) -> None:
        self._service = service

    def call(self, request: Request):
        email = request.data.get("email", "")
        name = request.data.get("name", "")
        password = request.data.get("password", "")

        response = self._service.call(email=email, name=name, password=password)

        return Response(
            status_code=200,
            request_id=request.request_id,
            action=request.action,
            data=response,
        )
